include/unified_teleop/controller.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef TELEOP_CONTROLLER_INCLUDE_GUARD_HPP
+#define TELEOP_CONTROLLER_INCLUDE_GUARD_HPP
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/joy.hpp"
+#include <optional>
+#include <map>
+#include <string>
+#include <cmath>
+#include "yaml-cpp/yaml.h"
+
+namespace rosnu
+{
+    enum class InputType
+    {
+        /// @brief  Represents an axis input, such as a joystick; the values usually start from 0.0 and range from -1.0 to 1.0.
+        Axis,
+        /// @brief Represents a trigger input, such as a trigger button; the values usually start from 1.0 and range from -1.0 to 1.0.
+        Trigger,
+        /// @brief Represents a button input, such as a regular button; the values are usually either 0 or 1.
+        Button,
+        None
+    };
+
+    /// @brief Represents an unused input type.
+    constexpr InputType UNUSED_TYPE = InputType::None;
+
+    /// @brief Represents an unused index value.
+    constexpr int UNUSED_INDEX = -1;
+
+    struct MovementInput
+    {
+        /// @brief Index number that correlates with its position in the joy message array
+        int index;
+        /// @brief InputType that indicates the type of input it is (Axis, Trigger, Button, None)
+        InputType type;
+
+        /// @brief Default constructor. Initializes index to UNUSED_INDEX and type to InputType::None
+        MovementInput() : index(UNUSED_INDEX), type(InputType::None) {}
+        /// @brief Constructor with provided index and type parameters.
+        /// @param index_no The index of the joy message array.
+        /// @param input_type The type of input.
+        MovementInput(int index_no, InputType input_type) : index(index_no), type(input_type) {}
+    };
+
+    class Controller
+    {
+        private:
+            sensor_msgs::msg::Joy current_joy_state, previous_joy_state;
+            bool always_enabled = false;
+            bool first_joy_state = true;
+            std::map<std::string, int> button_map;
+            std::vector<std::optional<rosnu::MovementInput>> MovementInput_vec;
+        
+        public:
+            //
```

Comment:
CONSTRUCTORS, FUNCTIONS etc are unnecessary comments and should be removed

Reply:
I thought they make it nicer to read :')